### ISSUE TITLE
[Bug]: cross compilation from linux to windows with build tools fails.

### ISSUE THREAD

**Issue by ogoffart** [corrosion-rs/corrosion#476]
### Current Behavior

corrosion adds ".exe" to the path of the tools it runs when it targets windows, but it shouldn't

### Expected Behavior

the tools runs on linux so it shouldn't add .exe

### Steps To Reproduce

This has been seen when trying to cross compile Slint application from linux that target windows.
More info there: https://github.com/slint-ui/slint/issues/2508

**Comment by ogoffart** [comment]
As per my comment on the Slint issue: https://github.com/slint-ui/slint/issues/2508#issuecomment-1893711655

https://github.com/corrosion-rs/corrosion/pull/437 fixes the other way around (windows to non-windows)
But the condition in https://github.com/corrosion-rs/corrosion/blob/8d8ff67e506cdee4a69123f3bea31b0465a48fe2/cmake/Corrosion.cmake#L81 doesn't cover the target windows but host not windows.

**Comment by jschwe** [comment]
changing the condition to this should be sufficient to fix this right?
```cmake 
if( (hostbuild AND CMAKE_HOST_WIN32) 
     OR 
     ((NOT hostbuild) AND (Rust_CARGO_TARGET_OS STREQUAL "windows")) )
```

**Comment by ogoffart** [comment]
yes, exactly.

**Comment by jschwe** [comment]
Fixed in v0.4.6
